[iOS] //flutter:unittests fails to compile for ios_debug_unopt

When building tests under `ios_debug_unopt` targeting physical device SDKs, we get an ambiguous method call error for `[mockSession configuration]` because `mockSession` was untyped (`id`), and the iOS device SDK includes `Metal.framework/Headers/MTL4PipelineDataSetSerializer.h`, which defines a `configuration` property returning a primitive enum type `MTL4PipelineDataSetSerializerConfiguration`.

Because the device SDK contains `configuration` methods returning both object pointers (such as `UISceneConfiguration*`) and primitive values, the compiler can't figure out the correct return type and calling convention for the invocation. This issue doesn't surface on simulator builds because `MTL4PipelineDataSetSerializer.h` header is stubbed out in the simulator SDK.

Even if we fix that, when building tests under `ios_debug_unopt` targeting device SDKs, the build fails at the linking stage with architecture compatibility errors. The `test_config` configuration in `//flutter/shell/platform/darwin/common/BUILD.gn` previously unconditionally appended `-mios-simulator-version-min`, regardless whether we were building for simulator or device.

### Compile errors
```
[macos/ios_debug_unopt: ninja] 50.0% (1/2) OBJCXX obj/flutter/she...s/framework/Source/libios_test_flutter.FlutterSceneLifeCycleTest.o[2026-07-16 08:23:28.490646][macos/ios_debug_unopt: ninja]: FAILED
COMMAND:
/Users/cbracken/Developer/flutter/flutter/third_party/ninja/ninja -C /Users/cbracken/Developer/flutter/flutter/engine/src/out/ios_debug_unopt -j 1000 flutter:unittests
STDOUT:
ninja: Entering directory `/Users/cbracken/Developer/flutter/flutter/engine/src/out/ios_debug_unopt'
FAILED: obj/flutter/shell/platform/darwin/ios/framework/Source/libios_test_flutter.FlutterSceneLifeCycleTest.o
../../flutter/buildtools/mac-arm64/clang/bin/clang++ -MMD -MF obj/flutter/shell/platform/darwin/ios/framework/Source/libios_test_flutter.FlutterSceneLifeCycleTest.o.d -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D_FORTIFY_SOURCE=2 -D_LIBCPP_DISABLE_AVAILABILITY=1 -D_LIBCPP_DISABLE_VISIBILITY_ANNOTATIONS -D_LIBCPP_ENABLE_THREAD_SAFETY_ANNOTATIONS -D_DEBUG -DSHELL_ENABLE_SOFTWARE -DSHELL_ENABLE_METAL -DSK_TYPEFACE_FACTORY_CORETEXT -DSK_FONTMGR_CORETEXT_AVAILABLE -DSK_CODEC_ENCODES_JPEG -DSK_CODEC_ENCODES_PNG -DSK_CODEC_ENCODES_PNG_WITH_LIBPNG -DSK_CODEC_ENCODES_WEBP -DSK_CODEC_DECODES_JPEG -DSK_CODEC_DECODES_PNG -DSK_CODEC_DECODES_ICO -DSK_CODEC_DECODES_PNG_WITH_LIBPNG -DSK_CODEC_DECODES_WEBP -DSK_HAS_WUFFS_LIBRARY -DSK_CODEC_DECODES_GIF -DSK_XML -DFLUTTER_RUNTIME_MODE_DEBUG=1 -DFLUTTER_RUNTIME_MODE_PROFILE=2 -DFLUTTER_RUNTIME_MODE_RELEASE=3 -DFLUTTER_RUNTIME_MODE_JIT_RELEASE=4 -DDART_LEGACY_API=\[\[deprecated\]\] -DFLUTTER_RUNTIME_MODE=1 -DFLUTTER_JIT_RUNTIME=1 -DSLIMPELLER=1 -DIMPELLER_DEBUG=1 -DIMPELLER_SUPPORTS_RENDERING=1 -DIMPELLER_ENABLE_METAL=1 -DSK_DISABLE_LEGACY_INIT_DECODERS -DSK_ENABLE_DUMP_GPU -DSK_LEGACY_IGNORE_DRAW_VERTICES_BLEND_WITH_NO_SHADER -DSK_DISABLE_LEGACY_MOCK_BACKENDSURFACE -DSK_DISABLE_LEGACY_SHADERCONTEXT -DSK_DISABLE_LOWP_RASTER_PIPELINE -DSK_FORCE_RASTER_PIPELINE_BLITTER -DSK_METAL_WAIT_UNTIL_SCHEDULED -DSK_DISABLE_EFFECT_DESERIALIZATION -DSK_ENABLE_PRECOMPILE -DSK_ASSUME_GL_ES=1 -DU_USING_ICU_NAMESPACE=0 -DU_ENABLE_DYLOAD=0 -DUSE_CHROMIUM_ICU=1 -DU_ENABLE_TRACING=1 -DU_ENABLE_RESOURCE_TRACING=0 -DU_STATIC_IMPLEMENTATION -DICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_FILE -DRAPIDJSON_HAS_STDSTRING -DRAPIDJSON_HAS_CXX11_RANGE_FOR -DRAPIDJSON_HAS_CXX11_RVALUE_REFS -DRAPIDJSON_HAS_CXX11_TYPETRAITS -DRAPIDJSON_HAS_CXX11_NOEXCEPT -DSK_ENABLE_PARAGRAPH -DSK_UNICODE_AVAILABLE -DSK_UNICODE_ICU_IMPLEMENTATION -DFLUTTER_API_SYMBOL_PREFIX=Embedder -DFLUTTER_NO_EXPORT -Igen/flutter/shell/platform/darwin/ios -I../.. -Igen -I../../flutter/third_party/libcxxabi/include -I../../flutter/build/secondary/flutter/third_party/libcxx/config -I../../flutter/prebuilts/Platforms/iPhoneOS.platform/Developer/usr/lib -Igen/flutter/shell/platform/darwin/common -I../../flutter -I../../flutter/third_party/ocmock/Source -I../../flutter/third_party/skia -I../../flutter/txt/src -I../../third_party/abseil-cpp -Igen/flutter -Igen/flutter/impeller/runtime_stage -I../../flutter/third_party/flatbuffers/include -I../../flutter/third_party/harfbuzz/src -I../../flutter/third_party/icu/source/common -I../../flutter/third_party/icu/source/i18n -I../../flutter/third_party/rapidjson -I../../flutter/third_party/rapidjson/include -I../../flutter/third_party -I../../flutter/third_party/dart/runtime -I../../flutter/third_party/dart/runtime/include -I../../flutter/third_party/skia/modules/skparagraph/include -I../../flutter/third_party/skia/modules/skunicode/include -I../../flutter/shell/platform/darwin/common/framework/Headers -I../../flutter/shell/platform/embedder -fvisibility=default -fno-strict-aliasing --target=arm64-apple-darwin -arch arm64 -fcolor-diagnostics -Wall -Wextra -Wendif-labels -Wno-missing-field-initializers -Wno-unused-parameter -Werror -Wno-unused-but-set-parameter -Wno-unused-but-set-variable -Wno-psabi -Wno-implicit-int-float-conversion -Wno-deprecated-copy -Wno-deprecated-literal-operator -Wno-unqualified-std-cast-call -Wno-nontrivial-memcall -Wno-non-c-typedef-for-linkage -Wno-range-loop-construct -Wunguarded-availability -Wno-deprecated-declarations -no-canonical-prefixes -fvisibility=hidden -Wstring-conversion -Wnewline-eof -O0 -g2 -fvisibility=default -mios-simulator-version-min=15.0 -F/Users/cbracken/Developer/flutter/flutter/engine/src/flutter/prebuilts/Platforms/iPhoneOS.platform/Developer/Library/Frameworks -Wunreachable-code -Wno-newline-eof  -Werror=overriding-method-mismatch -Werror=undeclared-selector -fobjc-arc -fvisibility-inlines-hidden -fobjc-call-cxx-cdtors -std=c++20 -Wno-deprecated-this-capture -fno-modules -Xclang -fno-cxx-modules -fno-rtti -nostdinc++ -I../../flutter/third_party/libcxx/include -I../../flutter/third_party/libcxxabi/include -nostdinc++ -fvisibility=hidden -fno-exceptions -fdebug-prefix-map=/Users/cbracken/Developer/flutter/flutter/engine/src/= -Wno-unguarded-availability -isysroot ../../flutter/prebuilts/SDKs/iPhoneOS26.5.sdk -miphoneos-version-min=15.0  -c ../../flutter/shell/platform/darwin/ios/framework/Source/FlutterSceneLifeCycleTest.mm -o obj/flutter/shell/platform/darwin/ios/framework/Source/libios_test_flutter.FlutterSceneLifeCycleTest.o
./engine/src/flutter/shell/platform/darwin/ios/framework/Source/FlutterSceneLifeCycleTest.mm:1051:11: error: multiple methods named 'configuration' found with mismatched result, parameter type or attributes
 1051 |   OCMStub([mockSession configuration]).andReturn(mockConfiguration);
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~
./engine/src/flutter/third_party/ocmock/Source/OCMock/OCMock.h:54:13: note: expanded from macro 'OCMStub'
   54 |             invocation; \
      |             ^~~~~~~~~~
./engine/src/flutter/third_party/ocmock/Source/OCMock/OCMock.h:161:5: note: expanded from macro '_OCMSilenceWarnings'
  161 |     macro \
      |     ^~~~~
./engine/src/flutter/prebuilts/SDKs/iPhoneOS26.5.sdk/System/Library/Frameworks/Foundation.framework/Headers/NSURLSession.h:141:55: note: one possibility
  141 | @property (readonly, copy) NSURLSessionConfiguration *configuration;
      |                                                       ^~~~~~~~~~~~~
./engine/src/flutter/prebuilts/SDKs/iPhoneOS26.5.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIImage.h:228:80: note: also found
  228 | @property (nullable, nonatomic, copy, readonly) __kindof UIImageConfiguration *configuration API_AVAILABLE(ios(13.0),tvos(13.0),watchos(6.0));
      |                                                                                ^~~~~~~~~~~~~
./engine/src/flutter/prebuilts/SDKs/iPhoneOS26.5.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIImageReader.h:39:67: note: also found
   39 | @property (nonatomic, readonly, copy) UIImageReaderConfiguration *configuration;
      |                                                                   ^~~~~~~~~~~~~
./engine/src/flutter/prebuilts/SDKs/iPhoneOS26.5.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIButton.h:68:73: note: also found
   68 | @property (nonatomic, readwrite, copy, nullable) UIButtonConfiguration *configuration API_AVAILABLE(ios(15.0), tvos(15.0)) API_UNAVAILABLE(watchos);
      |                                                                         ^~~~~~~~~~~~~
./engine/src/flutter/prebuilts/SDKs/iPhoneOS26.5.sdk/System/Library/Frameworks/UIKit.framework/Headers/UICollectionViewCompositionalLayout.h:63:77: note: also found
   63 | @property(nonatomic,copy) UICollectionViewCompositionalLayoutConfiguration *configuration;
      |                                                                             ^~~~~~~~~~~~~
./engine/src/flutter/prebuilts/SDKs/iPhoneOS26.5.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIContentConfiguration.h:35:56: note: also found
   35 | @property (nonatomic, copy) id<UIContentConfiguration> configuration;
      |                                                        ^~~~~~~~~~~~~
./engine/src/flutter/prebuilts/SDKs/iPhoneOS26.5.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIListContentConfiguration.h:128:57: note: also found
  128 | @property (nonatomic, copy) UIListContentConfiguration *configuration;
      |                                                         ^~~~~~~~~~~~~
./engine/src/flutter/prebuilts/SDKs/iPhoneOS26.5.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIFontPickerViewController.h:36:80: note: also found
   36 | @property (readonly, copy, nonatomic) UIFontPickerViewControllerConfiguration *configuration;
      |                                                                                ^~~~~~~~~~~~~
./engine/src/flutter/prebuilts/SDKs/iPhoneOS26.5.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIPasteControl.h:39:62: note: also found
   39 | @property (nonatomic, readonly) UIPasteControlConfiguration *configuration;
      |                                                              ^~~~~~~~~~~~~
./engine/src/flutter/prebuilts/SDKs/iPhoneOS26.5.sdk/System/Library/Frameworks/UIKit.framework/Headers/UISceneSession.h:27:61: note: also found
   27 | @property (nonatomic, readonly, copy) UISceneConfiguration *configuration;
      |                                                             ^~~~~~~~~~~~~
./engine/src/flutter/prebuilts/SDKs/iPhoneOS26.5.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIContentUnavailableView.h:26:64: note: also found
   26 | @property (nonatomic, copy) UIContentUnavailableConfiguration *configuration;
      |                                                                ^~~~~~~~~~~~~
./engine/src/flutter/prebuilts/SDKs/iPhoneOS26.5.sdk/System/Library/Frameworks/UIKit.framework/Headers/UITextFormattingViewController.h:71:84: note: also found
   71 | @property (nonatomic, readonly, copy) UITextFormattingViewControllerConfiguration *configuration;
      |                                                                                    ^~~~~~~~~~~~~
./engine/src/flutter/prebuilts/SDKs/iPhoneOS26.5.sdk/System/Library/Frameworks/Metal.framework/Headers/MTL4PipelineDataSetSerializer.h:50:76: note: also found
   50 | @property(readwrite, nonatomic) MTL4PipelineDataSetSerializerConfiguration configuration;
      |                                                                            ^~~~~~~~~~~~~
1 error generated.
ninja: build stopped: subcommand failed.

STDERR:


[2026-07-16T08:23:31.800][macos/ios_debug_unopt: RBE shutdown]: Actions completed: 0
```

### Linker errors
```
[2026-07-16 08:28:40.401916][macos/ios_debug_unopt: ninja]: FAILED
COMMAND:
/Users/cbracken/Developer/flutter/flutter/third_party/ninja/ninja -C /Users/cbracken/Developer/flutter/flutter/engine/src/out/ios_debug_unopt -j 1000 flutter:unittests
STDOUT:
ninja: Entering directory `/Users/cbracken/Developer/flutter/flutter/engine/src/out/ios_debug_unopt'
FAILED: libios_test_flutter.dylib libios_test_flutter.dylib.TOC
if [ ! -e ./libios_test_flutter.dylib -o ! -e ./libios_test_flutter.dylib.TOC ] || otool -l ./libios_test_flutter.dylib | grep -q LC_REEXPORT_DYLIB ; then ../../flutter/buildtools/mac-arm64/clang/bin/clang++ -shared -isysroot ../../flutter/prebuilts/SDKs/iPhoneOS26.5.sdk -miphoneos-version-min=15.0  -Wl,-object_path_lto,./lto_libios_test_flutter.o -Wl,-install_name,@rpath/Frameworks/libios_test_flutter.dylib --target=arm64-apple-darwin -arch arm64 -nostdlib++ -Wl,-unexported_symbols_list,../../flutter/third_party/libcxxabi/lib/new-delete.exp -mios-simulator-version-min=15.0 -L../../flutter/prebuilts/Platforms/iPhoneOS.platform/Developer/usr/lib -L../../flutter/prebuilts/SDKs/iPhoneOS26.5.sdk/usr/lib/swift -L../../flutter/prebuilts/SDKs/XcodeDefault.xctoolchain/usr/lib/swift/iphoneos -F../../flutter/prebuilts/Platforms/iPhoneOS.platform/Developer/Library/Frameworks -o ./libios_test_flutter.dylib -Wl,-filelist,./libios_test_flutter.dylib.rsp ./libocmock_shared.dylib -ldl -lpthread -framework XCTest -framework Foundation -framework CoreFoundation -framework ImageIO -framework MobileCoreServices -framework CoreGraphics -framework CoreText -framework UIKit -framework Metal -framework MetalPerformanceShaders -framework Security -framework CoreServices -framework AudioToolbox -framework CoreMedia -framework CoreVideo -framework IOSurface -framework QuartzCore -framework WebKit -framework IOKit -framework Testing && { otool -l ./libios_test_flutter.dylib | grep LC_ID_DYLIB -A 5; nm -gP ./libios_test_flutter.dylib | cut -f1-2 -d' ' | grep -v U$$; true; } > ./libios_test_flutter.dylib.TOC; else ../../flutter/buildtools/mac-arm64/clang/bin/clang++ -shared -isysroot ../../flutter/prebuilts/SDKs/iPhoneOS26.5.sdk -miphoneos-version-min=15.0  -Wl,-object_path_lto,./lto_libios_test_flutter.o -Wl,-install_name,@rpath/Frameworks/libios_test_flutter.dylib --target=arm64-apple-darwin -arch arm64 -nostdlib++ -Wl,-unexported_symbols_list,../../flutter/third_party/libcxxabi/lib/new-delete.exp -mios-simulator-version-min=15.0 -L../../flutter/prebuilts/Platforms/iPhoneOS.platform/Developer/usr/lib -L../../flutter/prebuilts/SDKs/iPhoneOS26.5.sdk/usr/lib/swift -L../../flutter/prebuilts/SDKs/XcodeDefault.xctoolchain/usr/lib/swift/iphoneos -F../../flutter/prebuilts/Platforms/iPhoneOS.platform/Developer/Library/Frameworks -o ./libios_test_flutter.dylib -Wl,-filelist,./libios_test_flutter.dylib.rsp ./libocmock_shared.dylib -ldl -lpthread -framework XCTest -framework Foundation -framework CoreFoundation -framework ImageIO -framework MobileCoreServices -framework CoreGraphics -framework CoreText -framework UIKit -framework Metal -framework MetalPerformanceShaders -framework Security -framework CoreServices -framework AudioToolbox -framework CoreMedia -framework CoreVideo -framework IOSurface -framework QuartzCore -framework WebKit -framework IOKit -framework Testing && { otool -l ./libios_test_flutter.dylib | grep LC_ID_DYLIB -A 5; nm -gP ./libios_test_flutter.dylib | cut -f1-2 -d' ' | grep -v U$$; true; } > ./libios_test_flutter.dylib.tmp && if ! cmp -s ./libios_test_flutter.dylib.tmp ./libios_test_flutter.dylib.TOC; then mv ./libios_test_flutter.dylib.tmp ./libios_test_flutter.dylib.TOC ; fi; fi
clang++: warning: using sysroot for 'iOS 26.5' but targeting 'arm64-apple-ios15.0.0-simulator' [-Wincompatible-sysroot]
ld64.lld: error: ../../flutter/prebuilts/SDKs/iPhoneOS26.5.sdk/usr/lib/libdl.tbd(/usr/lib/libSystem.B.dylib) is incompatible with arm64 (iOS Simulator15.0.0)
ld64.lld: error: ../../flutter/prebuilts/SDKs/iPhoneOS26.5.sdk/System/Library/Frameworks/Foundation.framework/Foundation.tbd(/System/Library/Frameworks/Foundation.framework/Foundation) is incompatible with arm64 (iOS Simulator15.0.0)
ld64.lld: error: ../../flutter/prebuilts/SDKs/iPhoneOS26.5.sdk/System/Library/Frameworks/CoreFoundation.framework/CoreFoundation.tbd(/System/Library/Frameworks/CoreFoundation.framework/CoreFoundation) is incompatible with arm64 (iOS Simulator15.0.0)
ld64.lld: error: ../../flutter/prebuilts/SDKs/iPhoneOS26.5.sdk/System/Library/Frameworks/ImageIO.framework/ImageIO.tbd(/System/Library/Frameworks/ImageIO.framework/ImageIO) is incompatible with arm64 (iOS Simulator15.0.0)
ld64.lld: error: ../../flutter/prebuilts/SDKs/iPhoneOS26.5.sdk/System/Library/Frameworks/MobileCoreServices.framework/MobileCoreServices.tbd(/System/Library/Frameworks/MobileCoreServices.framework/MobileCoreServices) is incompatible with arm64 (iOS Simulator15.0.0)
ld64.lld: error: ../../flutter/prebuilts/SDKs/iPhoneOS26.5.sdk/System/Library/Frameworks/CoreGraphics.framework/CoreGraphics.tbd(/System/Library/Frameworks/CoreGraphics.framework/CoreGraphics) is incompatible with arm64 (iOS Simulator15.0.0)
ld64.lld: error: ../../flutter/prebuilts/SDKs/iPhoneOS26.5.sdk/System/Library/Frameworks/CoreText.framework/CoreText.tbd(/System/Library/Frameworks/CoreText.framework/CoreText) is incompatible with arm64 (iOS Simulator15.0.0)
ld64.lld: error: ../../flutter/prebuilts/SDKs/iPhoneOS26.5.sdk/System/Library/Frameworks/UIKit.framework/UIKit.tbd(/System/Library/Frameworks/UIKit.framework/UIKit) is incompatible with arm64 (iOS Simulator15.0.0)
ld64.lld: error: ../../flutter/prebuilts/SDKs/iPhoneOS26.5.sdk/System/Library/Frameworks/Metal.framework/Metal.tbd(/System/Library/Frameworks/Metal.framework/Metal) is incompatible with arm64 (iOS Simulator15.0.0)
ld64.lld: error: ../../flutter/prebuilts/SDKs/iPhoneOS26.5.sdk/System/Library/Frameworks/MetalPerformanceShaders.framework/MetalPerformanceShaders.tbd(/System/Library/Frameworks/MetalPerformanceShaders.framework/MetalPerformanceShaders) is incompatible with arm64 (iOS Simulator15.0.0)
ld64.lld: error: ../../flutter/prebuilts/SDKs/iPhoneOS26.5.sdk/System/Library/Frameworks/Security.framework/Security.tbd(/System/Library/Frameworks/Security.framework/Security) is incompatible with arm64 (iOS Simulator15.0.0)
ld64.lld: error: ../../flutter/prebuilts/SDKs/iPhoneOS26.5.sdk/System/Library/Frameworks/CoreServices.framework/CoreServices.tbd(/System/Library/Frameworks/CoreServices.framework/CoreServices) is incompatible with arm64 (iOS Simulator15.0.0)
ld64.lld: error: ../../flutter/prebuilts/SDKs/iPhoneOS26.5.sdk/System/Library/Frameworks/AudioToolbox.framework/AudioToolbox.tbd(/System/Library/Frameworks/AudioToolbox.framework/AudioToolbox) is incompatible with arm64 (iOS Simulator15.0.0)
ld64.lld: error: ../../flutter/prebuilts/SDKs/iPhoneOS26.5.sdk/System/Library/Frameworks/CoreMedia.framework/CoreMedia.tbd(/System/Library/Frameworks/CoreMedia.framework/CoreMedia) is incompatible with arm64 (iOS Simulator15.0.0)
ld64.lld: error: ../../flutter/prebuilts/SDKs/iPhoneOS26.5.sdk/System/Library/Frameworks/CoreVideo.framework/CoreVideo.tbd(/System/Library/Frameworks/CoreVideo.framework/CoreVideo) is incompatible with arm64 (iOS Simulator15.0.0)
ld64.lld: error: ../../flutter/prebuilts/SDKs/iPhoneOS26.5.sdk/System/Library/Frameworks/IOSurface.framework/IOSurface.tbd(/System/Library/Frameworks/IOSurface.framework/IOSurface) is incompatible with arm64 (iOS Simulator15.0.0)
ld64.lld: error: ../../flutter/prebuilts/SDKs/iPhoneOS26.5.sdk/System/Library/Frameworks/QuartzCore.framework/QuartzCore.tbd(/System/Library/Frameworks/QuartzCore.framework/QuartzCore) is incompatible with arm64 (iOS Simulator15.0.0)
ld64.lld: error: ../../flutter/prebuilts/SDKs/iPhoneOS26.5.sdk/System/Library/Frameworks/WebKit.framework/WebKit.tbd(/System/Library/Frameworks/WebKit.framework/WebKit) is incompatible with arm64 (iOS Simulator15.0.0)
ld64.lld: error: ../../flutter/prebuilts/SDKs/iPhoneOS26.5.sdk/System/Library/Frameworks/IOKit.framework/IOKit.tbd(/System/Library/Frameworks/IOKit.framework/Versions/A/IOKit) is incompatible with arm64 (iOS Simulator15.0.0)
ld64.lld: error: ../../flutter/prebuilts/SDKs/iPhoneOS26.5.sdk/usr/lib/swift/libswiftUIKit.tbd(/usr/lib/swift/libswiftUIKit.dylib) is incompatible with arm64 (iOS Simulator15.0.0)
ld64.lld: error: too many errors emitted, stopping now (use --error-limit=0 to see all errors)
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.

STDERR:


[2026-07-16T08:28:42.419][macos/ios_debug_unopt: RBE shutdown]: Actions completed: 0
```

--- a/engine/src/flutter/shell/platform/darwin/common/BUILD.gn
+++ b/engine/src/flutter/shell/platform/darwin/common/BUILD.gn
@@ -39,13 +39,14 @@ config("test_config") {
       "$darwin_target_arch-apple-ios$ios_testing_deployment_target$darwin_target_sim",
     ]
     if (use_ios_simulator) {
-      cflags += [ "-mios-simulator-version-min=$ios_testing_deployment_target" ]
-      ldflags +=
-          [ "-mios-simulator-version-min=$ios_testing_deployment_target" ]
+      _version_min_flag =
+          "-mios-simulator-version-min=$ios_testing_deployment_target"
     } else {
-      cflags += [ "-miphoneos-version-min=$ios_testing_deployment_target" ]
-      ldflags += [ "-miphoneos-version-min=$ios_testing_deployment_target" ]
+      _version_min_flag =
+          "-miphoneos-version-min=$ios_testing_deployment_target"
     }
+    cflags += [ _version_min_flag ]
+    ldflags += [ _version_min_flag ]
   } else if (is_mac) {
     _platform_path = mac_platform_path
     cflags += [ "-mmacosx-version-min=$mac_testing_deployment_target" ]

--- a/engine/src/flutter/shell/platform/darwin/common/BUILD.gn
+++ b/engine/src/flutter/shell/platform/darwin/common/BUILD.gn
@@ -38,8 +38,14 @@ config("test_config") {
       "-target",
       "$darwin_target_arch-apple-ios$ios_testing_deployment_target$darwin_target_sim",
     ]
-    cflags += [ "-mios-simulator-version-min=$ios_testing_deployment_target" ]
-    ldflags += [ "-mios-simulator-version-min=$ios_testing_deployment_target" ]
+    if (use_ios_simulator) {
+      cflags += [ "-mios-simulator-version-min=$ios_testing_deployment_target" ]
+      ldflags +=
+          [ "-mios-simulator-version-min=$ios_testing_deployment_target" ]
+    } else {
+      cflags += [ "-miphoneos-version-min=$ios_testing_deployment_target" ]
+      ldflags += [ "-miphoneos-version-min=$ios_testing_deployment_target" ]
+    }
   } else if (is_mac) {
     _platform_path = mac_platform_path
     cflags += [ "-mmacosx-version-min=$mac_testing_deployment_target" ]

--- a/engine/src/flutter/shell/platform/darwin/ios/framework/Source/FlutterSceneLifeCycleTest.mm
+++ b/engine/src/flutter/shell/platform/darwin/ios/framework/Source/FlutterSceneLifeCycleTest.mm
@@ -1045,7 +1045,7 @@ FLUTTER_ASSERT_ARC
   OCMStub([mockRestorationPlugin restorationData]).andReturn(mockData);
 
   id mockScene = mocks[@"mockScene"];
-  id mockSession = OCMClassMock([UISceneSession class]);
+  UISceneSession* mockSession = OCMClassMock([UISceneSession class]);
   id mockConfiguration = OCMClassMock([UISceneConfiguration class]);
   OCMStub([mockScene session]).andReturn(mockSession);
   OCMStub([mockSession configuration]).andReturn(mockConfiguration);

